pkg/core/tokens/revocations.go

```diff
@@ -0,0 +1,66 @@
+package tokens
+
+import (
+	"context"
+	"strings"
+
+	system_proto "github.com/kumahq/kuma/api/system/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/resources/apis/system"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
+	core_store "github.com/kumahq/kuma/pkg/core/resources/store"
+)
+
+// Revocations keeps track of revoked tokens.
+// If only one token is compromised, it's more convenient to revoke it instead of rotate signing key and regenerate all tokens.
+// Revocation list is stored as Secret (in case of mesh scoped tokens) or GlobalSecret (global scoped tokens).
+// IDs of token are stored in secret in comma separated format: "id1,id2".
+type Revocations interface {
+	IsRevoked(id string) (bool, error)
+}
+
+func NewRevocations(manager manager.ReadOnlyResourceManager, revocationKey core_model.ResourceKey) Revocations {
+	return &secretRevocations{
+		manager:       manager,
+		revocationKey: revocationKey,
+	}
+}
+
+type secretRevocations struct {
+	manager       manager.ReadOnlyResourceManager
+	revocationKey core_model.ResourceKey
+}
+
+func (s *secretRevocations) IsRevoked(id string) (bool, error) {
```

Comment:
I'm wondering if potentially this will slow down the DPP start. Maybe we should keep the map of revoked tokens in memory, but I guess that's all about performance testing, not the scope of this PR